nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/main/java/org/nuxeo/ecm/platform/audit/io/LogEntryCSVWriter.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+package org.nuxeo.ecm.platform.audit.io;
+
+import static org.nuxeo.ecm.core.io.registry.reflect.Instantiations.SINGLETON;
+import static org.nuxeo.ecm.core.io.registry.reflect.Priorities.REFERENCE;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_CATEGORY;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_COMMENT;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_DOC_LIFE_CYCLE;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_DOC_PATH;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_DOC_TYPE;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_DOC_UUID;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_EVENT_DATE;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_EVENT_ID;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_ID;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_LOG_DATE;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_PRINCIPAL_NAME;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_REPOSITORY_ID;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ecm.core.io.marshallers.csv.AbstractCSVWriter;
+import org.nuxeo.ecm.core.io.registry.reflect.Setup;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+
+/**
+ * Convert {@link LogEntry} to CSV only keeping default and fetched properties if any.
+ *
+ * @since 11.1
+ */
+@Setup(mode = SINGLETON, priority = REFERENCE)
+public class LogEntryCSVWriter extends AbstractCSVWriter<LogEntry> {
+
+    private static final Logger log = LogManager.getLogger(LogEntryCSVWriter.class);
+
+    // FIXME to be unified with LogEntryJsonWriter#ENTITY_TYPE
+    public static final String ENTITY_TYPE = "logEntry";
+
+    public static final List<String> DEFAULT_PROPERTIES = List.of(LOG_ID, LOG_CATEGORY, LOG_PRINCIPAL_NAME, LOG_COMMENT,
+            LOG_DOC_LIFE_CYCLE, LOG_DOC_PATH, LOG_DOC_TYPE, LOG_DOC_UUID, LOG_EVENT_ID, LOG_REPOSITORY_ID,
+            LOG_EVENT_DATE, LOG_LOG_DATE);
+
+    @Override
+    protected void write(LogEntry entity, CSVPrinter printer) throws IOException {
+        List<String> propertiesToFetch = getPropertiesToFetch();
+        for (String property : propertiesToFetch) {
+            // getter names are consistent with property names. So they are easy to call by reflexion
+            String getterName = "get" + property.substring(0, 1).toUpperCase() + property.substring(1);
+            try {
+                printer.print(entity.getClass().getMethod(getterName).invoke(entity));
```

Comment:
Could you not use reflection? I know it's a pain to list all methods, but this is faster and doesn't lead to error/weird case for non existing properties.

Reply:
I did half of it before going for that solution. Hopefully saved it asside ;-)